### PR TITLE
CIRC-6524 - Host Resolution Race

### PR DESCRIPTION
**Pull request by pjulien**
- Proper race free host resolution
- For the fall back to `gethostname`, use a mutex, not
  perfect but will be better nonetheless

https://circonus.atlassian.net/browse/CIRC-6524

### CODE REVIEW

File: fq_client.c [hunk 1]
```diff
@@ -245,20 +245,47 @@ fq_resolve_endpoint(fq_conn_s *conn_s) {
     }
     memcpy(&conn_s->remote.sin_addr, *addr_list, sizeof(struct in_addr));
     free(buf);
+#elif defined _POSIX_C_SOURCE
+    struct addrinfo *result;
+
+    if (getaddrinfo(conn_s->host, NULL, NULL, &result)) {
```
Comment (vasuraman11) [inline fq_client.c:251]: no mutex needed here?

Comment (vasuraman11) [reply fq_client.c:251]: also, what about the first one?
`#ifdef HAVE_GETHOSTBYNAME_R`

Comment (pjulien) [reply fq_client.c:251]: These alternative APIs are thread safe, they don't use static data like `gethostbyname` does

**Review by vasuraman11 — APPROVED** [review]
LGTM

**Pull request merged**